lib/Runtime/Language/JavascriptOperators.h

```diff
@@ -80,6 +89,7 @@ namespace Js
 #endif
 
 #define BEGIN_TYPEOF_ERROR_HANDLER(scriptContext)  \
+    JavascriptExceptionObject* exceptionObject = nullptr; \
```

Comment:
Perhaps add another block here to scope the `exceptionObject` to these macros? I think you'd just need a `{` here, and a `}` in `END_TYPEOF_ERROR_HANDLER`